spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/dataAdapter/GeoJsonAdapter.java

```diff
@@ -58,13 +63,52 @@ public Observable connect() {
 
-                // if the file was packaged with the application, then attempt to connect
-                if (scStoreConfig.isMainBundle()) {
-                    // first check if the GeoJson file already exists in the internal storage location specified in the uri
-                    File f = new File(context.getFilesDir(), scStoreConfig.getUri());
+                if (scStoreConfig.getUri().startsWith("http")) {
+                    //download from web
+                    try {
+                        URL theUrl = new URL(scStoreConfig.getUri());
+                        HttpHandler.getInstance().get(theUrl.toString())
+                                .subscribe(new Action1<Response>() {
+                                    @Override
+                                    public void call(Response response) {
+                                        if (!response.isSuccessful()) {
+                                            adapterInstance.disconnect();
+                                        }
+                                        try {
+                                            // save response as file
+                                            File f = new File(context.getFilesDir(), scStoreConfig.getUri());
+                                            if (!f.exists()) {
+                                                FileUtils.copyInputStreamToFile(response.body().byteStream(), f);
+
+                                                adapterInstance.connected();
+                                                subscriber.onCompleted();
+                                            }
+                                        }
+                                        catch (IOException e) {
+                                            Log.w(LOG_TAG, "Couldn't download geojson store.", e);
+                                            adapterInstance.disconnect();
+                                            e.printStackTrace();
+                                            subscriber.onError(e);
+                                        }
+                                        finally {
+                                            response.body().close();
+                                        }
+                                    }
+                                });
+                    }
+                    catch (MalformedURLException e) {
+                        Log.e(LOG_TAG, "URL was malformed. Check the syntax: " + scStoreConfig.getUri());
+                        adapterInstance.setStatus(SCDataAdapterStatus.DATA_ADAPTER_DISCONNECTED);
+                        subscriber.onError(e);
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                } else {
+                    //attempt to find file local
+                    File f = new File(context.getFilesDir(), scStoreConfig.getUri().replace("file://", ""));
```

Comment:
Same question applies here....filename probably makes more sense as the id instead of the uri